apps/simple_history.py

```diff
@@ -0,0 +1,91 @@
+"""
+Copyright 2019 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+
+from django.core.serializers import serialize
+from django.db import models
+
+from simple_history.models import HistoricalRecords, ModelChange, ModelDelta
+
+
+RELATED_TO_LOCATION = ('shipment_to', 'shipment_from', 'shipment_dest', 'shipment_bill', )
+
+
+def get_user(request=None, instance=None):
+
+    if request:
+        return request.user.id
+
+    if instance:
+        if hasattr(instance, 'updated_by'):
+            return instance.updated_by
```

Comment:
I don't think this will return the correct user if a location is updated by a different user than the last user to update a shipment. Please add a unit test to cover this case to make sure that it works correctly.

Reply:
This clause is actually not necessary since we use the `simple_history_middleware` to retrieve the user in request.